src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -331,7 +331,7 @@ public Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library, NuGe
                 // If we found an exact match then use it.
                 // This allows us to shortcircuit slow feeds even if there's an exact match
                 if (!libraryRange.VersionRange.IsFloating &&
-                    match != null &&
+                    match?.Library != null &&
```

Comment:
I didn't debug this - I ran into null refs here when trying to get the msbuild task running. @davidfowl did make this change recently so I wanted to check if he already knew of this.